### ISSUE TITLE
Compare-at price doesn't show if varied across multiple variants

### ISSUE THREAD

**Issue by LexMetcalfe** [Shopify/dawn#871]
**Describe the current behavior**

The compare-at price will just show "Sale", and omit the dollar value of the compare at price if it is varied across a product's variants.

**Describe the expected behavior**

Have the compare at price to always regardless of variation across variants. 

**Version information (Dawn, browsers and operating systems)**
- Dawn Version: 2.2.0
- Chrome Version 95.0.4638.69
- macOS Version 12.01

**Possible solution**

Remove the price--no-compare class from the price `<div>` in price.liquid with a tickbox setting in the theme editor.

[Screenshot removed]

**Additional context/screenshots**

Before:
[Screenshot removed]

After:
[Screenshot removed]

**Comment by Roi-Arthur** [comment]
Adding a video since I think the screenshots used originally might have been hosted through Monosnap
https://user-images.githubusercontent.com/48017311/231120602-dba1d724-33d7-4903-850a-d78fa6a99a30.mp4

**Comment by randy-peterson** [comment]
Had a merchant today experiencing this issue on Sense 11.0. Applied Shawn's workaround above and that seemed to do the trick. Ticket: https://shopify.zendesk.com/agent/tickets/45560718

**Comment by Roi-Arthur** [comment]
Price is still the bane of our existence…
Would an approach like this work (to use less code)?
```liquid
{% assign cheapest_var = product.variants | sort: 'price' | first | default: product %}
{% if cheapest_var.compare_at_price > cheapest_var.price %}
<s>{{ cheapest_var.compare_at_price | money }}</s>
{% endif %}
{{ cheapest_var.price | money }}
```

The one case I can think of that might be ambiguous is if more than 1 variants have the same price equal to the lowest and have different compare_at_price. You'd then be returning the first variant with that price.

**Comment by Roi-Arthur** [comment]
Yeah I was trying to spin things some more but not really getting anywhere satisfying and not using several hugely loops/conditions.
We'll hopefully take a stab at simplifying price in the medium term

**Comment by mpickett-shopify** [comment]
This has been an issue for awhile now, and I see this has been closed. Not sure if we still need to look into it, but I had a merchant that pushed back on my suggested support options so I dug in with the VaultBot agent and, at least for the merchant I was working with, we were able to resolve the issue and get Compare at pricing to appear for products with the same base price price / same compare at price, same base price / different variant price. It's working on the Collection grid, product page as well as the other sections that display the product-card. 

We changed both of the instances in price.liquid of:

`{%- unless product.price_varies == false and product.compare_at_price_varies %}`

to 

`{%- unless product.price_varies == false and product.compare_at_price_varies and compare_at_price <= price %}`

In my testing I couldn't find any issues in either the display of pricing or add to cart functionality.
